quantestpy/circuit.py

```diff
@@ -17,34 +17,31 @@
 
 
 def assert_equal_to_operator(
+        circuit: Union[TestCircuit, str],
         operator_: Union[np.ndarray, np.matrix],
-        qasm: str = None,
-        qiskit_circuit=None,
-        test_circuit: TestCircuit = None,
         from_right_to_left_for_qubit_ids: bool = False,
         rtol: float = 0.,
         atol: float = 1e-8,
         up_to_global_phase: bool = False,
         msg=None) -> None:
 
-    if qasm is None and qiskit_circuit is None and test_circuit is None:
-        raise QuantestPyError(
-            "Missing qasm or qiskit_circuit or test circuit."
-        )
+    # test_circuit
+    if isinstance(circuit, TestCircuit):
```

Comment:
Line 28-44 appears in line 65-81 again. Can we make another function and use it?
For example, 
```python
def set_circuit(circuit: Union[TestCircuit, str]):

    if isinstance(circuit, TestCircuit):
        test_circuit = circuit
    # qasm
    elif isinstance(circuit, str):
        test_circuit = _cvt_openqasm_to_test_circuit(circuit)
    # qiskit.QuantumCircuit()
    elif _is_instance_of_qiskit_quantumcircuit(circuit):
        test_circuit = _cvt_qiskit_to_test_circuit(circuit)
    else:
        raise QuantestPyError(
            "Input circuit must be one of the following: "
            "qasm, qiskit.QuantumCircuit and TestCircuit."
        )
```

Reply:
@mitz1012 
Thank you! That is a great idea! 
I implemented it and now the code looks much simpler 👉 https://github.com/QuantestPy/quantestpy/pull/128/commits/27c6e176cf1cabe343378c5c4e831d18fb39ec45